modules/gwlb/variables.tf

```diff
@@ -1,69 +1,148 @@
+  - `private_ip_address_version`    - (`string`, optional, defaults to `IPv4`) the IP version for the private IP address.
+  EOF
+  nullable    = false
+  type = object({
+    name                       = string
+    subnet_id                  = string
+    private_ip_address         = optional(string)
+    private_ip_address_version = optional(string, "IPv4")
+  })
+  validation { # private_ip_address
+    condition = (var.frontend_ip.private_ip_address != null ?
+    can(regex("^(\\d{1,3}\\.){3}\\d{1,3}$", var.frontend_ip.private_ip_address)) : true)
+    error_message = "The `private_ip_address` property should be in IPv4 format."
+  }
+  validation { # private_ip_address_version
+    condition     = contains(["IPv4", "IPv6"], var.frontend_ip.private_ip_address_version)
+    error_message = "The `private_ip_address_version` property can be one of \"IPv4\", \"IPv6\"."
+  }
 }
 
 variable "health_probe" {
   description = <<-EOF
-  Health probe configuration for the gateway load balancer backends. Following settings are available:
-  - `name`                - (Optional|string) Name of the health probe. Defaults to `name` variable value.
-  - `port`                - (Required|int)
-  - `protocol`            - (Optional|string)
-  - `probe_threshold`     - (Optional|int)
-  - `request_path`        - (Optional|string)
-  - `interval_in_seconds` - (Optional|int)
-  - `number_of_probes`    - (Optional|int)
-  
-  For details, please refer to [provider documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_probe#argument-reference).
+  Health probe configuration for the Gateway Load Balancer backends.
+
+  Following settings are available:
+  - `name`                - (`string`, optional) name of the health probe. Defaults to `name` variable value.
+  - `port`                - (`number`, required) port to run the probe against.
+  - `protocol`            - (`string`, optional, defaults to `Tcp`) protocol used by the health probe,
+                            can be one of "Tcp", "Http" or "Https".
+  - `probe_threshold`     - (`number`, optional) number of consecutive probes that decide on forwarding traffic to an endpoint.
+  - `request_path`        - (`string`, optional) used only for non `Tcp` probes,
+                            the URI used to check the endpoint status when `protocol` is set to `Http(s)`.
+  - `interval_in_seconds` - (`number`, optional) interval in seconds between probes, with a minimal value of 5
+  - `number_of_probes`    - (`number`, optional) the number of failed probe attempts after which the backend endpoint
+                            is removed from rotation.
   EOF
-  type        = map(any)
+  default = {
+    port = 80
+  }
+  nullable = false
+  type = object({
+    name                = optional(string)
+    port                = number
+    protocol            = optional(string, "Tcp")
+    interval_in_seconds = optional(number, 15)
```

Comment:
I would also keep this the same as in LB module -> use Azure defaults

Reply:
do not use your own defaults, unless Azure ones do not make sense in our deployments -> less maintenance

Reply:
I removed default value for `protocol`, but I don't think that for LB we are following Azure defaults e.g.
- [interval_in_seconds](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_probe#interval_in_seconds) have default value 15, but in LB it's missed https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/ad1c4d7ff2cad59a59cb1fe3bf820e65a2586b88/modules/loadbalancer/variables.tf#L357
- [protocol](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_probe#protocol) is optional, not required as in https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/ad1c4d7ff2cad59a59cb1fe3bf820e65a2586b88/modules/loadbalancer/variables.tf#L354